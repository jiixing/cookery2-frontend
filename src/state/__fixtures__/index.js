import recipes from './recipes'
import * as rest from './rest'

const state = { recipes: recipes() }

export { recipes, rest }
export default () => state
