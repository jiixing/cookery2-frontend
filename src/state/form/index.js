import * as constants from './constants'

export { constants }
export { default as reducer } from './reducers'

export { modelName } from './utils'
