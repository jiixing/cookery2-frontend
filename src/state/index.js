import { combineReducers } from 'redux'

import {
  constants as recipesConstants,
  reducer as recipesReducer,
} from './recipes'
import {
  constants as recipeConstants,
  reducer as recipeReducer,
} from './recipe'

const rootReducer = combineReducers({
  [recipesConstants.namespace]: recipesReducer,
  [recipeConstants.namespace]: recipeReducer,
})

export default rootReducer
