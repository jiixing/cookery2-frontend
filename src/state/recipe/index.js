import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as recipeSelector } from './selectors'
export { default as reducer } from './reducers'
