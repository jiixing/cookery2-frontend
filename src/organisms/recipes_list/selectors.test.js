import state from 'state/__fixtures__'
import recipesSelector from './selectors'

describe('recipesSelector', () => {
  it('orders the books as categories', () => {
    expect(recipesSelector(state())).toMatchSnapshot()
  })
})
