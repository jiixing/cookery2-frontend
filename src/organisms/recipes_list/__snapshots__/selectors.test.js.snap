// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipesSelector orders the books as categories 1`] = `
Array [
  Object {
    "difficulty": "difficult",
    "duration": 0,
    "id": 5,
    "name": "Alpha King Pale Ale",
    "servings": 1,
  },
  Object {
    "difficulty": "easy",
    "duration": 0,
    "id": 2,
    "name": "Bell’s Expedition",
    "servings": 5,
  },
  Object {
    "difficulty": "medium",
    "duration": 0,
    "id": 4,
    "name": "Chimay Grande Réserve",
    "servings": 3,
  },
  Object {
    "difficulty": "medium",
    "duration": 0,
    "id": 3,
    "name": "Founders Kentucky Breakfast ",
    "servings": 2,
  },
  Object {
    "difficulty": "easy",
    "duration": 0,
    "id": 6,
    "name": "La Fin Du Monde",
    "servings": 6,
  },
  Object {
    "difficulty": "medium",
    "duration": 93,
    "id": 11,
    "name": "Samuel Smith’s Oatmeal Stout",
    "servings": 6,
  },
  Object {
    "difficulty": "difficult",
    "duration": 0,
    "id": 7,
    "name": "Trappistes Rochefort 10",
    "servings": 1,
  },
  Object {
    "difficulty": "medium",
    "duration": 0,
    "id": 10,
    "name": "Yeti Imperial Stout",
    "servings": 2,
  },
  Object {
    "difficulty": "easy",
    "duration": 0,
    "id": 15,
    "name": "my new thing",
    "servings": 3,
  },
  Object {
    "difficulty": "medium",
    "duration": 0,
    "id": 16,
    "name": "name name name",
    "servings": 3,
  },
  Object {
    "difficulty": "easy",
    "duration": 0,
    "id": 1,
    "name": "test name",
    "servings": 2,
  },
]
`;
